### PR TITLE
ci: prefer to set bazel repo cache in CI worker

### PR DESCRIPTION
**Pull request by wuhuizuo**
### What problem does this PR solve?

Prefer set Bazel repo cache args in CI worker rather than the git repo

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

- delete `--repository_cache=/home/jenkins/.tidb/tmp` bazel build arg in make task.
- prefer set it by `$HOME/.bazelrc` file in CI worker, it will be inherited when run `bazel build` or `bazel test`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -116,5 +116,5 @@ TEST_COVERAGE_DIR := "test_coverage"
 
 ifneq ("$(CI)", "0")
 	BAZEL_GLOBAL_CONFIG := --output_user_root=/home/jenkins/.tidb/tmp --host_jvm_args=-XX:+UnlockExperimentalVMOptions --host_jvm_args=-XX:+UseZGC
-	BAZEL_CMD_CONFIG := --config=ci --repository_cache=/home/jenkins/.tidb/tmp
+	BAZEL_CMD_CONFIG := --config=ci
```
Comment (xhebox) [inline Makefile.common:119]: If that can be done, what about `--output_user_root=/home/jenkins/.tidb/tmp`?

Comment (wuhuizuo) [reply Makefile.common:119]: All can be set in CI worker's `$HOME/.bazelrc`
- `--repository_cache` can be shared with different builds and workspaces.
- `--output_user_root` should not be shared with different workspaces.

The PR focus on optimize for shared repo cache, `--output_user_root` should be migrated in another PR.

**Review by xhebox — APPROVED** [review]

**Comment by hawkingrei** [comment]
Please don't remove it. CI must give control to developers.

**Comment by hawkingrei** [comment]
@wuhuizuo Your team is the most importance work to fix the remote cache. Why do you take time to fix the repository cache?

**Comment by hawkingrei** [comment]
1、fix the remote cache
2、put local cache and local repository_cache into the local disk. It is suggestion from bazel official team.

```
Note: We recommend you do not use an NFS or similar networked file system for the root directory, as the higher access latency will cause noticeably slower builds.
```

**Pull request closed without merging**